examples/stm32/f3/stm32f3-discovery/flash/flash.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2009 Uwe Hermann <uwe@hermann-uwe.de>
+ * Copyright (C) 2011 Stephen Caudle <scaudle@doceme.com>
+ * Modified by Fernando Cortes <fermando.corcam@gmail.com>
+ * modified by Guillermo Rivera <memogrg@gmail.com>
+ * modified by Roel Gerrits <roel@roelgerrits.nl>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/flash.h>
+
+static void gpio_setup(void)
+{
+	/* Enable GPIOE clock. */
+	rcc_periph_clock_enable(RCC_GPIOE);
+
+	/* Set GPIO12 (in GPIO port E) to 'output push-pull'. */
+	gpio_mode_setup(GPIOE, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE,
+            GPIO8 | GPIO9 | GPIO10 | GPIO11 | GPIO12 | GPIO13 | GPIO14 | GPIO15);
+}
+
+
+// this variable is made available by the linker script
+// it's address is the address where the my_flash section starts
+extern uint8_t _my_flash_start;
+
+
+int main(void)
+{
+	uint16_t counter;
+
+	gpio_setup();
+
+    // read value from flash
+    counter = _my_flash_start;
```

Comment:
Try running "make stylecheck" on your tree.  You need consistent tabs, and to use C style comments, not C++ comments.